Structure.Rmd

```diff
@@ -0,0 +1,398 @@
+  are subdirectories for 32 bit (`i386/`) and 64 bit (`x64/`) environments.
+
+* If you had any objects in `data/`, they have now been converted into a more
+  efficient form.
+  
+* The contents of `inst/` are moved to the top-level directory. For example,
+  vignette files are now in `doc/`.
+
+* Some files and folders have been dropped, such as `README`, `build/`, `tests/`,
+  and `vignettes/`.
+
+### Installed package
+
+An __installed__ package is a binary package that's been decompressed into a package library (described in \@ref(library)). The following diagram illustrates the many ways a package can be installed. This diagram is complicated! In an ideal world, installing a package would involve stringing together a set of simple steps: source -> bundle, bundle -> binary, binary -> installed. In the real world, it's not this simple because there are often (faster) shortcuts available.
+
+```{r, echo = FALSE}
+knitr::include_graphics("diagrams/installation.png")
+```
+
+The built-in command line tool `R CMD INSTALL` powers all package installation. It can install a package from source files, a bundle (a.k.a. a source tarball), or a binary package. Details are available in the [Installing packages section][ria-install] of [R Installation and Administration][ria]. Just like with `devtools::build()`, devtools provides a wrapper function, `devtools::install()`, that makes this tool available from within an R session.
+
+Most useRs understandably like to install packages from the comfort of an R session and directly from CRAN. The built-in function `install.packages()` meets this need. It can download the package, in various forms, install it, and optionally attend to the installation of dependencies.
+
+Devtools offers a family of `install_*()` functions to address some needs beyond the reach of `install.packages()` or to make existing capabilities easier to access:
+
+```{r}
+library(devtools)
+
+funs <- as.character(lsf.str("package:devtools"))
+grep("^install_.+", funs, value = TRUE)
+```
+
+`devtools::install_github()` is the flagship example of a sub-family of functions that can download a package from a remote location that is not CRAN and do whatever is necessary to install it. `install_github()` and friends are actually provided by the [remotes package](https://remotes.r-lib.org) and are re-exported by devtools. The rest of the devtools/remotes `install_*()` functions are aimed at making things that are technically possible with base tooling a bit easier or more explicit, such as `install_version()` which installs a specific version of a CRAN package.
+
+ Analogous to `.Rbuildignore`, described in section \@ref(rbuildignore), `.Rinstignore` lets you keep files present in a package bundle out of the installed package. However, in contrast to `.Rbuildignore`, this is rather obscure and rarely needed.
+
+### In-memory package
+
+*TODO: putting something here, so I can comment in my PR.*
```

Comment:
I haven't revised this section yet. I almost think it can go away?

I think all of this content is better off in the workflow sections (later in this chapter, next to be revised) and the chapter where `NAMESPACE` is handled. One way to look at it: all of the package forms addressed in "What is a package?" are defined in terms of the filesystem.

Nothing below here has changed, other than being involved in the rearrangement of sections.

Reply:
Agreed.  

Should we remove in-memory from the chapter all together?

Reply:
Kept it, to complete the story arc. But trimmed it down, with promises of greater coverage in those other places.